src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,137 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (isEmpty()) {
+            Node<T> node = new Node<>(value, null, null);
+            head = node;
+            tail = node;
+        } else {
+            Node<T> node = new Node<>(value, null, tail);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newNode = new Node<>(value, head, null);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        Node<T> node = getNode(index);
+        Node<T> newNode = new Node<>(value, node, node.prev);
+        node.prev.next = newNode;
+        node.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
+    }
+
+    private Node<T> getNode(int index) {
```

Comment:
let's move all private methods after public

Reply:
ok. moved it to the end & after privet class Node<T>